search/utils.py

```diff
@@ -176,10 +183,21 @@ def search_index(tokens=None, repo_slug=None, sort_by=None, terms=None):
         # Always sort by ID to preserve ordering.
         search = search.sort(sort_by, "id")
 
-    terms = set(get_vocab_names(repo_slug=repo_slug))
-    terms.update(set(('run', 'course', 'resource_type')))
+    terms = set(get_vocab_slugs(repo_slug=repo_slug))
     for term in terms:
-        search.aggs.bucket(term, "terms", field=term)
+        vocab_key = "vocab_{vocab}".format(vocab=term)
+        search.aggs.bucket(
+            "{term}_missing".format(term=vocab_key),
+            "missing", field=vocab_key
+        )
+        search.aggs.bucket(
+            "{term}_buckets".format(term=vocab_key),
+            "terms", field=vocab_key
+        )
+    for term in ('run', 'course', 'resource_type'):
+        search.aggs.bucket(
+            '{term}_builtins'.format(term=term), "terms", field=term
```

Comment:
The `_builtins` here is the label used in the aggregation, it isn't part of the index itself. In the index mapping vocabulary keys are prefixed with `vocab_` and other keys are the way they were before.